### PR TITLE
implement google geocoder

### PR DESCRIPTION
**Pull request by elliesterner**

### CODE REVIEW

File: google_geocoder.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+	if statusResponse.Status == GoogleGeocodeNoResult {
+		logrus.WithFields(logrus.Fields{
+			"Address": streetAddr,
+		}).Warn("Invalid address")
+
+		return nil, multierror.Append(err, errors.New("no results returned from google geocoder api"))
+	}
+
+	var apiResponse GoogleGeocodeApiResponse
```
Comment (c1freitas) [inline google_geocoder.go:152]: What is in this response? Can we abstract it into a non-google specific response object?

**Review by c1freitas — APPROVED** [review]
Minor comment. I also think it's worth adding a readme with the keys required and an example of how to use it.

File: google_geocoder.go (hunk 1 above)
Comment (elliesterner) [reply google_geocoder.go:152]: this is the generic lat/lng

type Coordinate struct {
	Lat  float64
	Long float64 `json:"lng"`
}

File: google_geocoder.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+const GoogleGeocodeNoResult = "ZERO_RESULTS" // message returned if no results are returned from google geocoder
+const GoogleGeocodeOKResult = "OK"           // message returned if valid results are returned from google geocoder
+
+type GoogleGeocoder struct {
+	BaseURL string
+	ApiKey  string
+}
+
+func NewGoogleGeocoder(baseURL string, apiKey string) Geocoder {
```
Comment (c1freitas) [inline google_geocoder.go:33]: Should this be called `NewGeocoder()` vs `NewGoogleGeoCoder` ?

Comment (c1freitas) [reply google_geocoder.go:33]: One point of an interface is to hide the implementation. If we are being explicit I am not sure of the value other than testing. which is valid.

Comment (elliesterner) [reply google_geocoder.go:33]: synced up with chuck on this! We agreed that NewGeocoder is best ... specifying the type in the function would only be helpful if we had multiple implementations. But right now we dont, and probably wont ever :)

Comment (elliesterner) [reply google_geocoder.go]: speaking of testing, I should do that.

**Pull request merged**